tck/features/SetAcceptance.feature

```diff
@@ -0,0 +1,332 @@
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A {property1: 23, property2: 46})
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      SET n.property1 = null
+      RETURN n
+      """
+    Then the result should be:
+      | n                    |
+      | (:A {property2: 46}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Setting a relationship property to null removes the existing property
+    Given any graph
+    And having executed:
+      """
+      CREATE ()-[:REL {property1: 12, property2: 24}]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r]->()
+      SET r.property1 = null
+      RETURN r
+      """
+    Then the result should be:
+      | r                      |
+      | [:REL {property2: 24}] |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A {name: 'Andres'})
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      WHERE n.name = 'Andres'
+      SET n.name = 'Michael'
+      RETURN n
+      """
+    Then the result should be:
+      | n                      |
+      | (:A {name: 'Michael'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property to an expression
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A {name: 'Andres'})
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      WHERE n.name = 'Andres'
+      SET n.name = n.name + ' was here'
+      RETURN n
+      """
+    Then the result should be:
+      | n                              |
+      | (:A {name: 'Andres was here'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property by selecting the node using a simple expression
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      SET (n).name = 'neo4j'
+      RETURN n
+      """
+    Then the result should be:
+      | n                    |
+      | (:A {name: 'neo4j'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property by selecting the node using an expression
+    Given any graph
+    And having executed:
+      """
+      CREATE (:A)
+      """
+    When executing query:
+      """
+      MATCH (n:A)
+      SET (CASE WHEN true THEN n END).name = 'neo4j'
+      RETURN n
+      """
+    Then the result should be:
+      | n                    |
+      | (:A {name: 'neo4j'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Set a property by selecting the relationship using a simple expression
+    Given any graph
+    And having executed:
+      """
+      CREATE ()-[:REL]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r:REL]->()
+      SET (r).name = 'neo4j'
```

Comment:
@Mats-SX yes, the original query is here: https://github.com/neo4j/neo4j/blob/3.1/community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/SetAcceptanceTest.scala#L104